ops/model.py

```diff
@@ -1445,6 +1445,9 @@ def set_content(self, content: Dict[str, str]):
         secret no longer exists, this method will succeed, but the unit will go
         into error state on completion of the current Juju hook.
 
+        Changed in Juju 3.6: setting the content to the same exact value no
+        longer creates a new secret revision.
+
```

Comment:
I think we must have had a discussion about adding this to the docs, since there's a Jira ticket, but I can't remember the details of that, sorry. It seems like this really belongs in the _Juju_ docs, not the _ops_ docs, because it's a Juju behaviour change. Like perhaps a note in the [secret lifecycle doc](https://juju.is/docs/juju/secret#heading--secret-lifecycle), and we link to that? Or [secret-set](https://juju.is/docs/juju/hook-tool#secret-set) and link to that?

On the other hand, if this opened with "This will create a new secret revision, if the content differs from the latest revision, and will notify all units ..." then I wouldn't have objected to that and thought it had too much Juju specific information. So maybe it is fine?

I think having a Sphinx directive for Juju changed as you hinted at in another comment would be better than just having it like this.

I don't love "same exact", maybe:

```suggestion
        Changed in Juju 3.6: setting the content with the same content as the latest
        revision longer creates a new secret revision.
```